examples/training_without_local_storage.ipynb

```diff
@@ -0,0 +1,917 @@
+    "                'bucket': s3_bucket_name,\n",
+    "                'prefix': bucket_prefix,\n",
+    "            },\n",
+    "            # In Jupyter, we set use_procs to False, since subprocess sometimes crash.\n",
+    "            # Outside of Jupyter, it is recommended to let use_procs default to True for performance \n",
+    "            use_procs=False,\n",
+    "        ),\n",
+    "        TensorboardLogger(log_dir=tensorboard_log_dir, flush_interval=1),\n",
+    "    ],\n",
+    "    save_folder=checkpoint_dir,\n",
+    "    save_interval=\"1ba\",\n",
+    "    save_num_checkpoints_to_keep=0,  # Because we are uploading checkpoints to the cloud, delete them locally to save disk space!\n",
+    "    run_name='cloud_training_run',\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Train!\n",
+    "trainer.fit()\n",
+    "\n",
+    "# Close the trainer, which will block on until all files have been uploaded\n",
+    "trainer.close()\n",
+    "\n",
+    "# Remove all local Tensorboard traces, since they were uploaded to the cloud\n",
+    "shutil.rmtree(tensorboard_log_dir, ignore_errors=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Great! We trained our model again, and this time, uploaded the checkpoints and Tensorboard logs to our S3 bucket. Let's verify that our files exist in the bucket."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "\n",
+    "s3 = boto3.client('s3')\n",
+    "\n",
+    "def print_objects_in_bucket(bucket: str, prefix: str):\n",
+    "    response = s3.list_objects(Bucket=bucket, Prefix=prefix)\n",
+    "    keys = [obj['Key'] for obj in response['Contents']]\n",
+    "    keys.sort()\n",
+    "    for k in keys:\n",
+    "        print(k)\n",
+    "\n",
+    "print_objects_in_bucket(s3_bucket_name, bucket_prefix)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's also take a look at visualizing tensorboard from our S3 Bucket:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import tensorboard.notebook\n",
+    "\n",
+    "%tensorboard --logdir s3://{s3_bucket_name}/{bucket_prefix}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can also resume training from a cloud checkpoint, without having to first download it. To do so, we'll need to set the `load_object_store` argument of the Trainer constructor to our `cloud_logger`, and the `load_path` to the object name of our checkpoint file within the bucket."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define the cloud logger\n",
```

Comment:
Perhaps define this earlier in the code so we don't have to rebuild the same logger multiple times

Reply:
Good point; putting them into helper functions.